jdbc-driver/src/main/java/com/google/cloud/dataproc/jdbc/DataprocDriverUtils.java

```diff
@@ -0,0 +1,246 @@
+                    try {
+                        port = Integer.valueOf(value);
+                    } catch (NumberFormatException e) {
+                        throw new InvalidURLException(
+                                String.format(
+                                        "Error: port=%s \n"
+                                                + "Please indicate correct port number or remove the field.",
+                                        value));
+                    }
+                    break;
+                // No need to parse value, already set in HiveJdbcParams
+                case "httpPath":
+                case "transportMode":
+                    break;
+
+                default:
+                    sb.append(";").append(pair);
+            }
+        }
+
+        if (sb.length() != 0) paramBuilder.setOtherSessionConfs(sb.toString());
+        paramBuilder.setPort(port);
+        // TODO: check if cluster name is valid / cluster master node name -m or -m-0
+        if (clusterName == null) {
+            throw new InvalidURLException(
+                    String.format("Error: clusterName=%s \n" + "Please indicate cluster name.", clusterName));
+        }
+
+        //        // Use this line when switched to supporting clusterPool
+        //        String hostName = !clusterName.isNull() ? clusterName : findClusterInPool(projectId,
+        // region, clusterPoolLabel);
+
+        paramBuilder.setHost(String.format("%s-m", clusterName));
+        return paramBuilder.build();
+    }
+
+    /**
+     * Format client passed in url that is accepted by Dataproc to be Hive acceptable format.
+     * jdbc:hive2://<host>:<port>/<dbName>;transportMode=http;httpPath=<http_endpoint>;<otherSessionConfs>?<hiveConfs>#<hiveVars>
+     *
+     * @param url client passed in url
+     * @return formatted url in Hive format
+     * @throws IOException
+     * @throws InvalidURLException
+     */
+    public static String formatHiveUrl(String url) throws InvalidURLException {
+        return parseHiveUrl(url).toHiveJdbcUrl();
+    }
+
+    /**
+     * Formats the user passed in clusterPoolLabel value to the filter format that Dataproc API
+     * accpets: field = value [AND [field = value]] ...
+     *
+     * @param filter user passed in str format: field1=value1:field2=value2
+     * @return formatted filter string accepted by Dataproc Client API
+     */
+    public static String formatClusterFilterString(String filter) {
+        return filter;
+    }
+
+    /**
+     * Supports picking a cluster from cluster pool.
+     *
+     * @param projectId client passed in project id
+     * @param region client passed in region to look for the cluster
+     * @param filter formatted filter that matches the conditions client passed in
+     * @return name of the suitable cluster with cluster pool
+     * @throws IOException
+     */
+    public static String findClusterInPool(String projectId, String region, String filter)
```

Comment:
You can omit this from this PR since it's not used/tested yet (and remove the commented line above)